source/extensions/filters/http/health_check/health_check.cc

```diff
@@ -38,6 +38,9 @@ Http::FilterHeadersStatus HealthCheckFilter::decodeHeaders(Http::HeaderMap& head
     health_check_request_ = true;
     callbacks_->requestInfo().healthCheck(true);
 
+    // Hint to tracer to ignore the span and any subsequent child spans
+    callbacks_->activeSpan().setSampled(false);
```

Comment:
Sorry I'm a little confused here. Why do we want to not sample health check requests? Can we make the comment more robust?

Reply:
As mentioned by @rshriram in https://github.com/envoyproxy/envoy/pull/3578#issuecomment-395902095, the health check probes generate a large number of unnecessary trace instances.

Previously the health checks were not being reported (assuming there were no child spans), because it didn't finish the span. So the behaviour is essentially the same from an Envoy perspective - however it fixes an issue in Istio as the mixer client performing the check is still being reported due to these health checks.

I will update the comment.